### PR TITLE
Updates the hardware page to add more detail

### PR DESCRIPTION
**Pull request by aturner-epcc**

### CODE REVIEW

**Comment by aturner-epcc** [comment]
Added more detail on processors in particular based on information from the hardware description on the NASA Aitkin description:

https://www.nas.nasa.gov/hecc/support/kb/amd-rome-processors_658.html

File: docs/user-guide/hardware.md [hunk 1]
```diff
@@ -52,12 +53,104 @@ More information on the architecture of the AMD EPYC Zen2 processors:
...
+    * Bandwidth: because the maximum throughput from the op cache is eight instructions per cycle, whereas the maximum throughput from the traditional fetch and decode pipeline is four instructions per cycle.
+    * Power: because there is no need to re-decode instructions.
+
+- L1 instruction cache: 32 KB, private to each core; 64 bytes/line; 8-way. The processor fetches instructions from the instruction cache in 32-byte naturally aligned blocks.
+- L1 data cache: 32 KB, private to each core; 64 bytes/line; 8-way; latency: 7–8 cycles for floating point and 4–5 cycles for integer; 2 x 256 bits/cycle load bandwidth to registers; 1 x 256 bits/cycle store bandwidth from registers; write-back policy.
+- L2 cache: 512 KB, private to each core; unified; inclusive of L1 cache; 64 bytes/line; 8-way; latency: >= 12 cycles; 1 x 256 bits/cycle load bandwidth to L1 cache; 1 x 256 bits/cycle store bandwidth from L1 cache; write-back policy.
+- L3 cache: 16 MB shared among four cores in a core complex (CCX); different CCXs do not share L3; total of 256 MB per socket. Within each CCX: 64 bytes/line; 16-way; latency: 39 cycles on average.
```
Comment (claoidek) [inline docs/user-guide/hardware.md:110]: This is the first place that the abbreviation CCX is defined, but it's used a few times above, so it should probably be defined earlier

File: docs/user-guide/hardware.md [hunk 2]
```diff
@@ -52,12 +53,104 @@ More information on the architecture of the AMD EPYC Zen2 processors:
...
+
+## Memory subsystem
+
+The Zen 2 microarchitecture places eight unified memory controllers in the centralized I/O die. The memory channels can be split into one, two, or four Non-Uniform Memory Access (NUMA) Nodes per Socket (NPS1, NPS2, and NPS4). ARCHER2 compute nodes are configured as NPS4, which is the highest memory bandwidth configuration geared toward HPC applications.
+
+With eight 3,200-GHz memory channels, an 8-byte read or write operation taking place per cycle per channel results in a maximum total memory bandwidth of 204.8 GB/s per socket.
+
+Each memory channel can be connected with up to two Double Data Rate (DDR) fourth-generation Dual In-line Memory Modules (DIMMs). On ARCHER2 standard memory nodes, each channel is connected to a single 16-GB DDR4 registered DIMM (RDIMM) with error correcting code (ECC) support leading to 128 GB per socket and 256 GB per node. For the high memory nodes, each channel is connected to a single 32-GB DDR4 registered DIMM (RDIMM) with error correcting code (ECC) support leading to 256 GB per socket and 512 GB per node.
```
Comment (claoidek) [inline docs/user-guide/hardware.md:136]: Should the hyphen in 32-GB be there?

Comment (aturner-epcc) [reply docs/user-guide/hardware.md]: Well spotted, should be fixed now

**Review by claoidek — APPROVED** [review]
LGTM

**Pull request merged**